packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,263 @@
+import * as React from "react";
+import {
+  K8sResourceCommon,
+  ListPageBody,
+  ListPageCreateLink,
+  ListPageFilter,
+  RowProps,
+  TableColumn,
+  TableData,
+  useActiveColumns,
+  useK8sWatchResource,
+  useListPageFilter,
+  VirtualizedTable
+} from "@openshift-console/dynamic-plugin-sdk";
+import { K8sModel } from "@openshift-console/dynamic-plugin-sdk/lib/api/common-types";
+import { TFunction } from "i18next";
+import * as _ from "lodash";
+import { useTranslation } from "react-i18next";
+import { sortable } from "@patternfly/react-table";
+import { BucketClassType, NS_PROVIDER_REVERSE_MAP } from "../../constants";
+import {
+  NooBaaBackingStoreModel,
+  NooBaaBucketClassModel,
+  NooBaaNamespaceStoreModel
+} from "../../models";
+import { BucketClassKind, NamespaceStoreKind } from "../../types";
+import { referenceForModel } from "../../utils";
+import { OperandStatus } from "../../utils/generics/operand-status";
+import { Kebab } from "../../utils/kebab/kebab";
+import {
+  LaunchModal,
+  ModalMap,
+  useModalLauncher
+} from "../../utils/modals/modalLauncher";
+import { getName } from "../../utils/selectors/k8s";
+import "./list-page.scss";
+import { bucketClassResource, nameSpaceStoreResource } from "../resources";
+
+const tableColumnInfo = [
+  { className: "", id: "name" },
+  {
+    className: "",
+    id: "status"
+  },
+  {
+    className: "",
+    id: "provider"
+  },
+  {
+    className: "",
+    id: "buckets"
+  },
```

Comment:
no class name? Do you want to display all 4 fields on all screen sizes?

Reply:
these are not mentioned, currently displaying all. later we can have it as an enhancement

Reply:
If we can add it in this PR, that would be great!  Also if you know something you'll do in follow up please add them as content or create tasks otherwise we will never be able to track it up.

Reply:
Sure @cloudbehl, even I updated the class names and handled it in this PR only.